### ISSUE TITLE
[performance] Tune the latency performance

### ISSUE THREAD

**Issue by clockfly** [intel-hadoop/gearpump#743]

**Comment by manuzhang** [comment]
As per the [discussion on akka-user group](https://groups.google.com/d/msg/akka-user/vWHHTDx6AUw/Gv8kJ-MJnf8J) and [akka scheduler documentation](http://doc.akka.io/docs/akka/snapshot/scala/scheduler.html) , akka scheduler executes (over)due tasks at every tick (configured by `akka.scheduler.tick-duration`, 10ms be default). This explains why I've witnessed mean receive latency of 20ms under not high load. 

@clockfly 
1. How about setting the tick duration to 1ms ? 
2. Another issue is our netty conf netty.fulsh-check-interval is not used. Is it supposed to used in the duration value below (which is now fixed at 5ms) ? 

```scala
  def messageHandler : Receive = {
    case msg: TaskMessage =>
      batch += msg
    case flush @ Flush(flushChannel)  =>
      if (channel != flushChannel) {
        Unit //Drop, as it belong to old channel flush message
      } else if (batch.size > 0 && flushChannel.isWritable) {
        send(flushChannel, batch.iterator)
        batch.clear()
        self ! flush
      } else {
        import context.dispatcher
        context.system.scheduler.scheduleOnce(new FiniteDuration(5, TimeUnit.MILLISECONDS), self, flush)
      }
  }

```

**Comment by clockfly** [comment]
Good finding!

+1

**Comment by clockfly** [comment]
Seems the is fixed by PR #785

**Comment by clockfly** [comment]
Close this.
